From: Vitaly Mayatskikh <vitaly@redhat.com>
Date: Thu, 16 Aug 2007 17:42:15 +0200
Subject: [acpi] sbs: file permissions set incorrectly
Message-id: m3zm0rh4fc.fsf@dhcp-lab-143.englab.brq.redhat.com
O-Subject: [RHEL5 patch] BZ242565 SBS parameter file permissions set incorrectly
Bugzilla: 242565

BZ#242565

https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=242565

Description:
Driver sbs set incorrect permissions of sysfs-files.

Upstream Status of the patch:
Kernel 2.6.22 has this issue fixed, patch backproted to RHEL5

Test Status:
It has been tested on x86.

$ ls -l /sys/module/sbs/parameters/
total 0
-r--r--r-- 1 root root 4096 2007-08-16 17:38 capacity_mode
-rw-r--r-- 1 root root 4096 2007-08-16 17:38 update_info_mode
-rw-r--r-- 1 root root 4096 2007-08-16 17:38 update_mode
-rw-r--r-- 1 root root 4096 2007-08-16 17:38 update_time
-rw-r--r-- 1 root root 4096 2007-08-16 17:38 update_time2

You can get patched kernel here:
http://brewweb.devel.redhat.com/brew/taskinfo?taskID=924285

Acked-by: Eric Sandeen <sandeen@redhat.com>
Acked-by: Prarit Bhargava <prarit@redhat.com>
---
 drivers/acpi/sbs.c |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/acpi/sbs.c b/drivers/acpi/sbs.c
index 62bef0b..b474f27 100644
--- a/drivers/acpi/sbs.c
+++ b/drivers/acpi/sbs.c
@@ -98,11 +98,11 @@ static int update_info_mode = UPDATE_INFO_MODE;
 static int update_time = UPDATE_TIME;
 static int update_time2 = UPDATE_TIME2;
 
-module_param(capacity_mode, int, CAPACITY_UNIT);
-module_param(update_mode, int, UPDATE_MODE);
-module_param(update_info_mode, int, UPDATE_INFO_MODE);
-module_param(update_time, int, UPDATE_TIME);
-module_param(update_time2, int, UPDATE_TIME2);
+module_param(capacity_mode, int, 0444);
+module_param(update_mode, int, 0644);
+module_param(update_info_mode, int, 0644);
+module_param(update_time, int, 0644);
+module_param(update_time2, int, 0644);
 
 static int acpi_sbs_add(struct acpi_device *device);
 static int acpi_sbs_remove(struct acpi_device *device, int type);
-- 
1.5.3.5.645.gbb47

